h5netcdf/core.py

```diff
@@ -384,6 +379,40 @@ def __repr__(self):
         )
 
 
+def _get_padding(h5ds, shape, key):
```

Comment:
The idea here is to really avoid creating new references to the h5ds accidentally through the `self` object. All references should be created before hand. This actually speeds things up more than I wish it would.